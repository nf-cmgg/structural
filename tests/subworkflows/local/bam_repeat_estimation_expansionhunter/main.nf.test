nextflow_workflow {

    name "Test Workflow BAM_REPEAT_ESTIMATION_EXPANSIONHUNTER"
    script "subworkflows/local/bam_repeat_estimation_expansionhunter/main.nf"
    workflow "BAM_REPEAT_ESTIMATION_EXPANSIONHUNTER"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_repeat_estimation_expansionhunter"

    test("homo_sapiens") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                input[3] = Channel.value([
                    [id:"catalog"],
                    file(params.expansionhunter_catalog, checkIfExists:true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path(workflow.out.vcfs[0][1]).linesGzip[-4..-1],
                    file(workflow.out.vcfs[0][2]).name,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

}
