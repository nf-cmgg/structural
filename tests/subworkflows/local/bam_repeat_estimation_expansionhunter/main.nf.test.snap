{
    "homo_sapiens": {
        "content": [
            [
                "chrX\t12000\t.\tT\t.\t.\tLowDepth\tEND=50000;REPREF=9500;RL=38000;RU=CAGG;VARID=CNBP;REPID=CNBP\tGT:SO:REPCN:REPCI:ADSP:ADFL:ADIR:LC\t.:.:.:.:.:.:.:0",
                "chrX\t12000\t.\tT\t.\t.\tLowDepth\tEND=123456;REPREF=18576;RL=111456;RU=GGCCCC;VARID=C9ORF72;REPID=C9ORF72\tGT:SO:REPCN:REPCI:ADSP:ADFL:ADIR:LC\t.:.:.:.:.:.:.:0",
                "chrX\t14586\t.\tG\t.\t.\tLowDepth\tEND=35698;REPREF=5278;RL=21112;RU=CAGA;VARID=CNBP_CAGA;REPID=CNBP_CAGA\tGT:SO:REPCN:REPCI:ADSP:ADFL:ADIR:LC\t.:.:.:.:.:.:.:0",
                "chrX\t15000\t.\tA\t.\t.\tLowDepth\tEND=45826;REPREF=15413;RL=30826;RU=CA;VARID=CNBP_CA;REPID=CNBP_CA\tGT:SO:REPCN:REPCI:ADSP:ADFL:ADIR:LC\t.:.:.:.:.:.:.:0"
            ],
            "test.expansionhunter.vcf.gz.tbi",
            3,
            0
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.03.0"
        },
        "timestamp": "2024-05-06T15:05:05.525988275"
    }
}